RocketChat/Rocket.Chat
# [NEW] REST endpoint to delete a DM and allow DM for two other users

## Proposed changes

I have a problem where I need the ability for the API to start (create) and stop (delete) DMs. The current im.create action allows specifying multiple usernames, however it also includes the calling user in the DM meaning that a bot user will also end up included in the DM.

The use-case difference between im.close and a proposed im.delete is that im.delete would remove the DM from all parties, instead of just hiding for the current user.

Both the change to allow creation of DMs excluding the API user and deleting of DMs require the `view-room-administration` permission given:
* to make the create action change make sense we need to also "open" the DMs for all participants. No doubt this is likely not something that you'd want a regular user to be able to do (potentially down for abuse)
* to call the delete action it required changing the "find" permission to allow third parties to look up a DM room. Again, to protect this action from abuse

## Issue(s)
Implements https://github.com/RocketChat/feature-requests/issues/353

## How to test or reproduce

This PR modifies the `im.create` API to support an optional `excludeSelf` boolean parameter. When set to true it will exclude the calling user from the created DM, allowing bot users to create DMs on behalf of users. To test, with a user assign a role with the `view-room-administration` permission POST the `usernames` parameter (specifying at least 2 other usernames) as well as `"excludeSelf": true`. A DM room with only the specified users should be created.

If you send the `excludeSelf` attribute but do not have the required `view-room-administration` permission it will treat the request as if `excludeSelf` was false.

This PR also adds a new `im.delete` API to allow deletion of DMs. This would also be used by bot users on behalf of users to remove existing DMs. To test, with a user assign a role with the `view-room-administration` permission POST the DM room ID parameter. The room should be deleted.

If posting without the `view-room-administration` permission the request will return a 403 forbidden.

## Types of changes




## Checklist



## Changelog

[NEW] Improve DM create/delete API management


## Further comments
